feat(module): handle `v5`

XanderD99 opened the pull request:
Resolves nuxt-modules/strapi#423

## Types of changes
- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)


## Description
a very basic v5 implementation. If I read their docs correctly only the returned format seems to have changed so I added that in a new v5 composable

It does needs to be tested however

Commit:
add basic v5

benjamincanac reviewed (changes requested):
Thanks for looking into this!

1. You can remove the unnecessary `package-lock.json` as we're using `pnpm`
2. You don't need to split the composable for `v5`, you can just have a `useStrapi`, this was implemented like this to prevent breaking changes.

benjamincanac reviewed (commented):
You might also need to export the `v5` types: https://github.com/nuxt-modules/strapi/blob/dev/src/runtime/types/index.d.ts#L574

Commit:
update types export

Commit:
remove package-lock.json

Commit:
add package-lock.json to gitignore

Commit:
update v5 types

Commit:
add v5 option in version options

Commit:
update useStrapi for v5

XanderD99 commented:
So updated according to your comments.
Added the types, as well as remove duplicates that remained the same from v4 => v5
Delete the package-lock and also added it to gitignore so it won't be added again in the future (I don't use pnpm :( )
Updated the split implementation (hope what I did was what you ment). But also updated some variable names, for example id => documentId to reflect the changed of strapi v5

benjamincanac commented on src/runtime/composables-v5/useStrapi.ts:
```diff
@@ -0,0 +1,101 @@
+export const useStrapi = <T>(): StrapiV5Client<T> => {
+  const client = useStrapiClient()
+  const version = useStrapiVersion()
+  if (version !== 'v5') {
```
Not sure you need this check now.

benjamincanac commented on src/runtime/composables-v5/useStrapi.ts:
```diff
@@ -0,0 +1,101 @@
+
+interface StrapiV5Client<T> {
+  find<F = T>(contentType: string, params?: Strapi5RequestParams): Promise<Strapi5ResponseMany<F>>
+  findOne<F = T>(contentType: string, docuemntId?: string | Strapi5RequestParams, params?: Strapi5RequestParams): Promise<Strapi5ResponseSingle<F>>
```
Why not keep the old signature here? The `id?: string | number | Strapi5RequestParams` seems fine to me.

XanderD99 replied:
changed it to documentId (don't mind the typo, will fix that) because Strapi v5 is moving away from using id as name and it is now documentId instead.

I wanted to reflect that change here as well. To indicate it want a documentId and not just an id.

benjamincanac commented on src/runtime/types/v5.d.ts:
```diff
@@ -0,0 +1,42 @@
+  locale?: StrapiLocale
+}
+
+export type Strapi5ResponseData<T> = Strapi5SystemFields & T
```
Not sure you want to remove the `Strapi5Response` as it might be used by users.

XanderD99 replied:
Removed it as it was duplicate from Strapi5ResponseSingle

Commit:
remove unneeded check in v5

4very commented:
`documentId` is misspelled in a couple of places as `docuemntId`.

Thank you for doing this, I'm excited to use v5 with Nuxt.

Commit:
fix documentId typos

benjamincanac committed:
Merge branch 'dev' into pr/424

benjamincanac committed:
up

benjamincanac commented:
I"m merging this so people can try it out using the [Edge](https://strapi.nuxtjs.org/setup#edge) package but I'd like some feedback before releasing it officially 😊

BayBreezy commented:
Hey. Thanks for the work guys. I use the old version of this on v5 of strapi. It still worked well lol
I usually just use a custom client with the useStrapiClient composable instead of the find, update methods that the package provide. I will give this a shot soon
